### PR TITLE
Add recapitation, and make it the default.

### LINKED ISSUE #259: [slim-engine-dev] Support recap/neutral-mutation-overlay with pyslim
There is some commented out code for this already, but some details need to be figured out. I think this should be the default, with a command line parameter to turn it off and do within-slim burn-in.

### PR DESCRIPTION
**Pull request by grahamgower**
Closes #259.

### CODE REVIEW

**Comment by grahamgower** [comment]
I don't see any RNG comments. Did this get lost somehow?

File: stdpopsim/slim_engine.py [hunk 1]
```diff
@@ -625,41 +638,28 @@ def script_file_f():
 
             ts = pyslim.load(ts_file.name)
 
-        # random.seed(seed)
-        # s1, s2 = random.randint(1,2**32-1), random.randint(1,2**32-1)
+        if do_recap:
+            random.seed(seed)
```
Comment (jeromekelleher) [inline stdpopsim/slim_engine.py]: Should create our own instance of RNG here rather than modifying the global state, ``rng = random.Random(seed)``.

**Comment by jeromekelleher** [comment]
> I don't see any RNG comments. Did this get lost somehow?

Hmm, did get lost, ta.

File: stdpopsim/slim_engine.py (hunk 1 above)
Comment (grahamgower) [reply stdpopsim/slim_engine.py]: Done. I also changed `randint(1, 2**32-1)` to the much clearer `randrange(2**32)`.

File: stdpopsim/slim_engine.py [hunk 2]
```diff
@@ -625,41 +638,28 @@ def script_file_f():
 
             ts = pyslim.load(ts_file.name)
 
-        # random.seed(seed)
-        # s1, s2 = random.randint(1,2**32-1), random.randint(1,2**32-1)
+        if do_recap:
+            rng = random.Random(seed)
+            s1, s2 = rng.randrange(2**32), rng.randrange(2**32)
```
Comment (jeromekelleher) [inline stdpopsim/slim_engine.py]: randrange doesn't work I'm afraid, as 0 isn't a valid seed for msprime.

Comment (jeromekelleher) [reply stdpopsim/slim_engine.py]: Well, randrange does work, you just need to put in ``rng.randrange(1, 2**32)``

**Pull request merged**